pandas/core/groupby.py

```diff
@@ -2742,74 +2741,124 @@ def _convert_grouper(axis, grouper):
         return grouper
 
 
-def _whitelist_method_generator(klass, whitelist):
-    """
-    Yields all GroupBy member defs for DataFrame/Series names in _whitelist.
-
-    Parameters
-    ----------
-    klass - class where members are defined.  Should be Series or DataFrame
-
-    whitelist - list of names of klass methods to be constructed
-
-    Returns
-    -------
-    The generator yields a sequence of strings, each suitable for exec'ing,
-    that define implementations of the named methods for DataFrameGroupBy
-    or SeriesGroupBy.
-
-    Since we don't want to override methods explicitly defined in the
-    base class, any such name is skipped.
-    """
-
-    method_wrapper_template = \
-        """def %(name)s(%(sig)s) :
-    \"""
-    %(doc)s
-    \"""
-    f = %(self)s.__getattr__('%(name)s')
-    return f(%(args)s)"""
-    property_wrapper_template = \
-        """@property
-def %(name)s(self) :
-    \"""
-    %(doc)s
-    \"""
-    return self.__getattr__('%(name)s')"""
-    for name in whitelist:
-        # don't override anything that was explicitly defined
-        # in the base class
-        if hasattr(GroupBy, name):
-            continue
-        # ugly, but we need the name string itself in the method.
-        f = getattr(klass, name)
-        doc = f.__doc__
-        doc = doc if type(doc) == str else ''
-        if isinstance(f, types.MethodType):
-            wrapper_template = method_wrapper_template
-            decl, args = make_signature(f)
-            # pass args by name to f because otherwise
-            # GroupBy._make_wrapper won't know whether
-            # we passed in an axis parameter.
-            args_by_name = ['{0}={0}'.format(arg) for arg in args[1:]]
-            params = {'name': name,
-                      'doc': doc,
-                      'sig': ','.join(decl),
-                      'self': args[0],
-                      'args': ','.join(args_by_name)}
-        else:
-            wrapper_template = property_wrapper_template
-            params = {'name': name, 'doc': doc}
-        yield wrapper_template % params
-
-
+# TODO: *args/**kwargs get ignored in many of the methods
+# inherited from Series/DataFrame.  Should they be used, dropped ...?
 class SeriesGroupBy(GroupBy):
-    #
-    # Make class defs of attributes on SeriesGroupBy whitelist
     _apply_whitelist = _series_apply_whitelist
-    for _def_str in _whitelist_method_generator(Series,
-                                                _series_apply_whitelist):
-        exec(_def_str)
+
+    @Appender(Series.all.__doc__)
+    def all(self, axis=None, bool_only=None, skipna=None, level=None,
+            **kwargs):
+        f = self.__getattr__('all')
+        return f(axis=axis, bool_only=bool_only, skipna=skipna, level=level)
+
+    @property
+    def dtype(self):
+        """
+         return the dtype object of the underlying data
+        """
+        return self.__getattr__('dtype')
```

Comment:
this indirection is extremely odd to do, simply use ``getattr(self.obj, 'dtype')``